hstream-store/HStream/Store/Stream.hs

```diff
@@ -545,6 +546,22 @@ listStreamPartitionsOrdered client streamid = do
   V.sortBy (\e1 e2 -> compare (snd e1) (snd e2)) mvec
   V.unsafeFreeze mvec
 
+-- Sorted by log name
+listStreamPartitionsOrderedByName
+  :: HasCallStack
+  => FFI.LDClient
+  -> StreamId
+  -> IO (Vector (CBytes, FFI.C_LogID))
+listStreamPartitionsOrderedByName client streamid = do
+  dir_path <- getStreamDirPath streamid
+  keys <- LD.logDirLogsNames =<< LD.getLogDirectory client dir_path
+  ps <- forM (V.fromList keys) $ \key -> do
+    logId <- getUnderlyingLogId client streamid (Just key)
+    pure (key, logId)
+  !mvec <- V.unsafeThaw ps
+  V.sortBy (\e1 e2 -> compare (fst e1) (fst e2)) mvec
+  V.unsafeFreeze mvec
```

Comment:
> Why not just sort by keys first 

Because of performance. `sort` on List is slower than on Vector.

> avoid call thaw and freeze

`unsafeThaw` and `unsafeFreeze` have no effect on performance downgrade.